pysb/pattern.py

```diff
@@ -63,7 +57,42 @@ def _get_bonds_in_monomer_pattern(mp):
     else:
         raise ValueError('Unknown pattern type: %s' % type(pat))
 
-    return set(bonds_used) if as_set else bonds_used
+    return bonds_used
+
+
+def get_bonds_in_pattern(pat):
+    """
+    Return the set of integer bond numbers used in a pattern
+
+    To return as a list (with duplicates), use
+    :func:`get_half_bonds_in_pattern`
+
+    Parameters
+    ----------
+    pat : ComplexPattern, MonomerPattern, or None
+        A pattern from which bond numberings are extracted
+
+    Returns
+    -------
+    set
+        Bond numbers used in the supplied pattern
+
+    Examples
+    --------
+
+    These examples convert the set to a list for testing purposes only (sets
```

Comment:
Oof, this is unfortunate. What about writing the examples slightly differently to sidestep the issue? Maybe with equality tests:
```python
>>> get_bonds_in_pattern(A(b1=None, b2=None)) == set()
True
>>> get_bonds_in_pattern(A(b1=1) % A(b2=1)) == {1}
True
...
```